client/my-sites/site-settings/form-seo.jsx

```
@@ -0,0 +1,279 @@
+				yandex: get( site, 'options.verification_services_codes.yandex', '' )
+			}
+		};
+	},
+
+	handleMetaChange( event ) {
+		this.setState( { seoMetaDescription: get( event, 'target.value', '' ) } );
+	},
+
+	handleVerificationCodeChange( event, serviceName ) {
+		const servicesCodes = this.state.verificationServicesCodes;
+		if ( ! servicesCodes.hasOwnProperty( serviceName ) ) {
+			return;
+		}
+
+		servicesCodes[ serviceName ] = get( event, 'target.value', '' );
+
+		this.setState( { verificationServicesCodes: servicesCodes } );
+	},
+
+	submitSeoForm( event ) {
+		const site = this.props.site;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		this.setState( { submittingForm: true } );
+
+		const filteredCodes = omitBy( this.state.verificationServicesCodes, isBoolean );
+		const updatedOptions = {
+			seo_meta_description: this.state.seoMetaDescription,
+			verification_services_codes: filteredCodes
+		};
+
+		site.saveSettings( updatedOptions, error => {
+			if ( error ) {
+				switch ( error.error ) {
+					case 'invalid_ip':
+						notices.error( this.translate( 'One of your IP Addresses was invalid. Please, try again.' ) );
+						break;
+					default:
+						notices.error( this.translate( 'There was a problem saving your changes. Please, try again.' ) );
+				}
+				this.setState( { submittingForm: false } );
+			} else {
+				notices.success( this.translate( 'Settings saved!' ) );
+				this.markSaved();
+				this.setState( { submittingForm: false } );
+
+				site.fetchSettings();
+			}
+		} );
+
+		analytics.tracks.recordEvent( 'calypso_seo_settings_form_submit' );
+	},
+
+	getGeneralSettingsUrl( slug ) {
+		return `/settings/general/${ slug }`;
+	},
+
+	render() {
+		const { slug = '', settings: { blog_public = 1 } = {}, fetchingSettings } = this.props.site;
+		const { submittingForm, seoMetaDescription, verificationServicesCodes } = this.state;
```

could we spit these into multiple lines for the sake of reader-clarity? especially with the nested defaults, I think this can be misleading or difficult

```js
const {
	slug = '',
	settings: {
		blog_public = 1
	} = {},
	fetchingSettings,
} = this.props;

const {
	fetchingSettings: isFetchingSettings, // rename to indicate function?
	setMetaDescription,
	submittingForm,
	verificationServicesCodes,
} = this.state;
```